File: SmartApp/app/src/main/java/cn/cmy/smartapp/persenter/LoginPresenter.java
package cn.cmy.smartapp.persenter;

import cn.cmy.smartapp.model.LoginModel;
import cn.cmy.smartapp.model.LoginResultCallBack;
import cn.cmy.smartapp.view.LoginView;

//p层
//特点1：持有m层的引用
//特点2：持有v层的引用
//特点3：对m层和v层进行关联
public class LoginPresenter extends BasePresenter<LoginView> {

    private LoginModel loginModel;

    public LoginPresenter() {
        loginModel = new LoginModel();
    }

    public void login(String userName, String password) {
        loginModel.login(userName, password, new LoginResultCallBack() {
            @Override
            public void loginResult(String result) {
                if (getV() != null) {
                    getV().onResult(result);
                }
            }
        });
    }


}

File: mvp/presenter/LoginPresenter.kt
package com.cmy.mvp.presenter

import com.cmy.mvp.model.LoginCallback
import com.cmy.mvp.model.LoginModel
import com.cmy.mvp.view.ILoginView
import java.lang.Exception

class LoginPresenter : BasePresenter<ILoginView>() {

    var model: LoginModel? = null

    init {
        model = LoginModel()
    }

    fun login(name: String, pwd: String) {
        model?.login(name, pwd, object : LoginCallback {
            override fun onSuccess(msg: String) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
                v?.onLoginResult(msg)
            }

            override fun onError(e: Exception) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
                v?.onLoginResult(e.message.toString())
            }
        })
    }


}